libs/agent-sm/agent-policy/src/main/java/org/opensearch/secure_sm/policy/PolicyFile.java

```diff
@@ -0,0 +1,346 @@
+
+    private void init(URL url) throws PolicyInitializationException {
+        PolicyInfo newInfo = new PolicyInfo();
+        initPolicyFile(newInfo, url);
+        policyInfo = newInfo;
+    }
+
+    private void initPolicyFile(final PolicyInfo newInfo, final URL url) throws PolicyInitializationException {
+        init(url, newInfo);
+    }
+
+    private void init(URL policy, PolicyInfo newInfo) throws PolicyInitializationException {
+        try (InputStreamReader reader = new InputStreamReader(getInputStream(policy), StandardCharsets.UTF_8)) {
+            PolicyParser policyParser = new PolicyParser();
+            policyParser.read(reader);
+
+            for (GrantEntry grantEntry : policyParser.grantElements()) {
+                addGrantEntry(grantEntry, newInfo);
+            }
+
+        } catch (Exception e) {
+            throw new PolicyInitializationException("Failed to load policy from: " + policy, e);
+        }
+    }
+
+    public static InputStream getInputStream(URL url) throws IOException {
+        if ("file".equals(url.getProtocol())) {
+            String path = url.getFile().replace('/', File.separatorChar);
+            path = URLDecoder.decode(path, StandardCharsets.UTF_8);
+            return new FileInputStream(path);
+        } else {
+            return url.openStream();
+        }
+    }
+
+    private CodeSource getCodeSource(GrantEntry grantEntry, PolicyInfo newInfo) throws PolicyInitializationException {
+        try {
+            Certificate[] certs = null;
+            URL location = (grantEntry.codeBase != null) ? newURL(grantEntry.codeBase) : null;
+            return canonicalizeCodebase(new CodeSource(location, certs));
+        } catch (Exception e) {
+            throw new PolicyInitializationException("Failed to get CodeSource", e);
+        }
+    }
+
+    private void addGrantEntry(GrantEntry grantEntry, PolicyInfo newInfo) throws PolicyInitializationException {
+        CodeSource codesource = getCodeSource(grantEntry, newInfo);
+        if (codesource == null) {
+            throw new PolicyInitializationException("Null CodeSource for: " + grantEntry.codeBase);
+        }
+
+        PolicyEntry entry = new PolicyEntry(codesource);
+        List<PermissionEntry> permissionList = grantEntry.permissionElements();
+        for (PermissionEntry pe : permissionList) {
+            expandPermissionName(pe);
+            try {
+                Optional<Permission> perm = getInstance(pe.permission, pe.name, pe.action);
+                if (perm.isPresent()) {
+                    entry.add(perm.get());
+                }
+            } catch (ClassNotFoundException e) {
+                // these were mostly custom permission classes added for security
+                // manager. Since security manager is deprecated, we can skip these
+                // permissions classes.
+                if (PERM_CLASSES_TO_SKIP.contains(pe.permission)) {
+                    continue; // skip this permission
```

Comment:
this is bit controversial, but better than eating expectations. Once we trim policies post JSM deprecation, we will no longer need this.